compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -378,101 +380,137 @@ typeOfTmLam (var, typ) body = do
 typeOfTyLam :: MonadGamma m => (TypeVarName, Kind) -> Expr -> m Type
 typeOfTyLam (tvar, kind) expr = TForall (tvar, kind) <$> introTypeVar tvar kind (typeOf expr)
 
-typeOfAlts :: MonadGamma m => (CaseAlternative -> m Type) -> [CaseAlternative] -> m Type
+-- | Type to track which constructor ranks have be covered in a pattern matching.
+data MatchedRanks = AllRanks | SomeRanks !(Set.Set Int)
```

Comment:
probably doesn’t matter much here but `IntSet` is almost always faster than `Set Int` ime.

Reply:
I thought about using `IntSet` but my impression was that it is not suited for small sets, which we'll have to deal with here most of the time. Something like `Data.BitSet` seems to be ideal for the use case we have here. Maybe plain lists and a `nubSort` at the end are already good enough? It's really hard to tell without proper benchmarking. But I'm not sure that worth our attention right now.

Reply:
It’s not that bad for small sets but yeah I agree on not trying to guess this without benchmarking which seems a waste of time atm.

Reply:
I just checked the implementation of `IntSet`, after @sofiafaro-da pointed out the same thing, and I was positively surprised. I thought it would be roughly the same as `IntMap ()` but it's actually a lot smarter. If all members of a set are smaller than 64 (on a 64-bit machine), then it pretty much amounts to using a 64-bit int as a bit set. I'll thus use `IntSet`, which seems the best from an unbenchmarked point of view.